lib/ansible/parsing/utils/yaml.py

```diff
@@ -43,20 +43,23 @@ def _handle_error(json_exc, yaml_exc, file_name, show_content):
     raise AnsibleParserError(n_err_msg, obj=err_obj, show_content=show_content, orig_exc=yaml_exc)
 
 
-def _safe_load(stream, file_name=None, vault_secrets=None):
+def _safe_load(stream, file_name=None, vault_secrets=None, all_docs=False):
     ''' Implements yaml.safe_load(), except using our custom loader class. '''
 
     loader = AnsibleLoader(stream, file_name, vault_secrets)
     try:
-        return loader.get_single_data()
+        if all_docs:
+            return loader.get_data()
```

Comment:
```suggestion
            docs = []
            while loader.check_data():
                docs.append(loader.get_data())
            return docs
```

Without this change, `get_data` would act just like `get_single_data`.  And unfortunately using `yield` here, instead of making a lis, causes oddities resulting in pickling errors.

FYI, this is needed as a dependency to hopefully support playbook argspecs.